sdk/tests/utilities/test_api_client_factory_auth_flow.py

```diff
@@ -0,0 +1,119 @@
+import unittest
+import os
+from lusid import InstrumentsApi, ResourceListOfInstrumentIdTypeDescriptor
+from lusid.utilities import ApiClientFactory
+from utilities import CredentialsSource
+from unittest.mock import patch
+
+source_config_details, config_keys = CredentialsSource.fetch_credentials(), CredentialsSource.fetch_config_keys()
+
+
+class ApiFactory(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        # create a configured API client
+        cls.secrets = CredentialsSource.secrets_path()
+        cls.os_environ_dict_str = 'os.environ'
+
+    def validate_api(self, api):
+        result = api.get_instrument_identifier_types()
+        self.assertIsNotNone(result)
+        self.assertIsInstance(result, ResourceListOfInstrumentIdTypeDescriptor)
+        self.assertGreater(len(result.values), 0)
+
+    def get_pat_env_var(self):
+
+        pat_env_vars = {
+            "FBN_LUSID_API_URL": os.getenv("FBN_LUSID_API_URL"),
```

Comment:
these will cause tests to fail (locally) if using a secrets file, might need to do something like https://github.com/finbourne/lusid-sdk-python-preview/blob/master/sdk/tests/utilities/credentials_source.py#L31

Reply:
Agreed. Updated.